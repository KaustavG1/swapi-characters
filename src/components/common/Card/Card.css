.character-card {
  border: 1px solid black;
  height: 250px;
  width: 200px;
  border-radius: 5px;
  cursor: pointer;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  transition: all .2s ease-in-out;
  box-shadow: 5px 5px;
}

.character-card:hover {
  transform: scale(1.05);
}

.card-name {
  padding-bottom: 20px;
  font-size: 1.2rem;
}

.card-details {
  padding: 20px 5px;
}
