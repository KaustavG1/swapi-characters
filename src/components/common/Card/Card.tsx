import { useNavigate } from "react-router";
import { Character } from "../../../models/Character";
import "./Card.css";

export interface CardProps {
  character: Character;
}

function Card({ character }: CardProps) {
  const navigate = useNavigate();

  const handleCardClick = () => {
    navigate(`/${character?.uid}`);
  };

  return (
    <div className="character-card" onClick={handleCardClick}>
      <div className="card-name">{character?.name}</div>
      <div className="card-details">Gender: {character?.gender}</div>
      <div className="card-details">Home Planet: {character?.homeworld}</div>
    </div>
  );
}

export default Card;
