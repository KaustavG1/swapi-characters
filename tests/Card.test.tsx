import { it, expect, vi } from "vitest";
import { fireEvent, render, screen } from "@testing-library/react";
import Card from "../src/components/common/Card/Card";

describe("Card component", () => {
  it("renders 'Card' component", () => {
    const onClick = vi.fn();
    const { container } = render(
      <Card
        name="Test Name"
        gender="Male"
        homeworld="Test Planet"
        onClick={onClick}
      />
    );
    const card = screen.getByText(/Test Name/i);
    fireEvent.click(card);
    expect(onClick).toHaveBeenCalledOnce();
    expect(card).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
