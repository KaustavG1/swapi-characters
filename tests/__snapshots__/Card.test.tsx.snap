// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card component > renders 'Card' component 1`] = `
<div>
  <div
    class="character-card"
  >
    <div
      class="card-name"
    >
      Test Name
    </div>
    <div
      class="card-details"
    >
      Gender: 
      Male
    </div>
    <div
      class="card-details"
    >
      Home Planet: 
      Test Planet
    </div>
  </div>
</div>
`;
